app/src/main/java/fr/free/nrw/commons/upload/FileMetadataUtils.java

```diff
@@ -0,0 +1,102 @@
+package fr.free.nrw.commons.upload;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import io.reactivex.Observable;
+import timber.log.Timber;
+
+import static androidx.exifinterface.media.ExifInterface.*;
+
+public class FileMetadataUtils {
+
+    public static Observable<String> getTagsFromPref(String pref) {
+        Timber.d("Retuning tags for pref:%s", pref);
+        switch (pref) {
+            case "Author":
+                return Observable.fromArray(TAG_ARTIST, TAG_CAMARA_OWNER_NAME);
+            case "Copyright":
+                return Observable.fromArray(TAG_COPYRIGHT);
+            case "Location":
+                return Observable.fromArray(TAG_GPS_LATITUDE, TAG_GPS_LATITUDE_REF,
+                        TAG_GPS_LONGITUDE, TAG_GPS_LONGITUDE_REF,
+                        TAG_GPS_ALTITUDE, TAG_GPS_ALTITUDE_REF);
+            case "Camera Model":
+                return Observable.fromArray(TAG_MAKE, TAG_MODEL);
+            case "Lens Model":
+                return Observable.fromArray(TAG_LENS_MAKE, TAG_LENS_MODEL, TAG_LENS_SPECIFICATION);
+            case "Serial Numbers":
+                return Observable.fromArray(TAG_BODY_SERIAL_NUMBER, TAG_LENS_SERIAL_NUMBER);
+            case "Software":
+                return Observable.fromArray(TAG_SOFTWARE);
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Removes all XMP data from the input file and writes the rest of the image to a new file.
+     *
+     * This works by black magic. Please read the JPEG section of the XMP Specification Part 3 before making changes.
```

Comment:
IMO, we should not add any method that works by black magic. :)

The attached doc is of 80 pages. Can you specify which part of the doc is relevant?

Reply:
Unfortunately, I can't. I agree with @ilgazer that it is better to move XMP handling in a separate PR as it needs testing.